dsl/jvm/src/main/kotlin/io/mockk/InternalPlatformDsl.kt

```diff
@@ -232,3 +257,26 @@ class JvmCoroutineCall<T>(private val lambda: suspend () -> T) : CoroutineCall<T
         }
     }
 }
+
+// TODO this is copy-pasted from ValueClassSupport
+//      I will try to move that class so it's available here
+
+private val valueClassFieldCache = mutableMapOf<KClass<out Any>, KProperty1<out Any, *>>()
+
+private fun <T : Any> KClass<T>.valueField(): KProperty1<out T, *> {
+    @Suppress("UNCHECKED_CAST")
+    return valueClassFieldCache.getOrPut(this) {
+        require(isValue) { "$this is not a value class" }
+
+        // value classes always have a primary constructor...
+        val constructor = primaryConstructor!!.apply { isAccessible = true }
+        // ...and exactly one constructor parameter
+        val constructorParameter = constructor.parameters.first()
+        // ...with a backing field
+        val backingField = declaredMemberProperties
+            .first { it.name == constructorParameter.name }
+            .apply { isAccessible = true }
+
+        backingField
+    } as KProperty1<out T, *>
+}
```

Comment:
I'm not sure how to resolve this.

* `ValueClassSupport.kt` is in project `mockk-agent-jvm`
* `InternalPlatformDsl` is in project `mockk-dsl-jvm`

I'm looking for suggestions. Maybe it's okay just to have it copy and pasted?

Reply:
yep i think if they are in those two separate modules there's no better way and we can live with it being copypasted.

Reply:
I think a comment where the same functionality is implemented would help, i case that code ever needs to be updated.

Reply:
Good idea, and I've made an issue to try and de-duplicated it later #857